marklogic-data-hub/src/main/java/com/marklogic/hub/HubProject.java

```diff
@@ -247,6 +247,13 @@
      */
     void upgradeProject() throws IOException;
 
+    /**
+     * Exports the project content to disk
+     *
+     * @throws IOException if there are issues with exporting project to disk.
+     */
+    void exportProject() throws IOException;
```

Comment:
While the story calls for a path, it's a Gradle-specific path, and I don't think our Java layer should know what it is. Therefore, I think this method should accept a File (or Path) as an input.

Reply:
Also, let's not throw an IOException here - IMO, IOException shouldn't be a checked exception because 99% of the time, there's nothing the caller can do to recover. No checked exception should be thrown, we'll rethrow checked ones in the implementation.

Reply:
The story doesn't mention a path to be passed to "hubExportProject" task . It says that the exported file should be available at "./build/datahub-project.zip". So the story should be changed  so that "hubExportProject" accepts a path (something like hubExportProject -Ppath=<path_export_file>)

Reply:
What I mean is that the Gradle task should know the path of "build/datahub-project.zip" and pass that in as a File to the Java class. The Java class shouldn't know about a "build" directory, since that's a Gradle-specific construct. It's an implementation detail that isn't relevant to the user story, but it matters in that we don't want the Java class knowing about Gradle-specific stuff.